src/2025h1/improve-rustc-codegen.md

```diff
@@ -0,0 +1,120 @@
+# Improve rustc codegen
+
+<!--
+
+> **Instructions:** Copy this template to a fresh file with a name based on your plan.
+> Give it a title that describes what you plan to get done in the next 6 months
+> (e.g., "stabilize X" or "nightly support for X" or "gather data about X").
+> Feel free to replace any text with anything, but there are placeholders
+> designed to help you get started. 
+
+-->
+
+| Metadata |                                                              |
+| -------- | ------------------------------------------------------------ |
+| Owner(s) | @folkertdev, @bjorn3                                         |
+| Teams    | T-compiler                                                   |
+| Status   | Proposed                                                     |
+
+## Summary
+
+We want to improve rustc codegen, based on [this initialive](https://trifectatech.org/initiatives/workplans/codegen/) by the Trifecta Tech Foundation. The work focusses on improving state machine code generation, and finding (and hopefully fixing) cases where clang produces better code than rustc for roughly equivalent input.
+
+## Motivation
+
+Matching C performance is crucial for rust adoption in performance-sensitive domains. Rust is doing well overall, but not good enough. 
+
+In compression, video decoding and other high-performance areas, nobody will use rust if it is even a couple percent slower: latency, power (i.e. battery) consumption and other factors are just more important than whatever advantages rust can bring. In particular, we've observed that C code translated to rust code, whether manually or mechanically, often performs a couple percent worse than the original C.
+
+Given that we clang and rustc both use LLVM for code generation, there is no fundamental reason that rust should be slower.
```

Comment:
I can think of at least one fairly fundamental difference that slightly disadvantages rust compared to C++: the (lack of) forward progress guarantee.